proposals/online-migrations-poc-and-proposal.md

```diff
@@ -161,23 +161,27 @@ IDs. Can we have conflicts if an insert is performed simultaneously in
 
 Another way to tackle this problem could be with the combination of:
 
-1. Non-breaking DB changes between two immediate Fleet versions, so both Fleet
-   versions can use the same database.
+1. Ensuring that two immediate Fleet versions can use the same database without
+   problems, this involves developer effort and planning. For example you can't
+   drop a column between two immediate Fleet versions, to do so you need a
+   release between that stops using the column altogether (more details
+   below.)
 2. [MySQL online DDL][mysql-ddl], which lets you to write non-locking alter
    statements via `ALTER ... LOCK=NONE`. This functionality is not available
    for certain operations (notably changing a column data type) but they're
    very few, all have a workaround, and per `1` we don't want immediate
    breaking changes anyway.
 
-Common scenarios (based on our latest 10 migrations):
+Common scenarios (based on the latest 10 migrations):
 
-- Adding a new column: as long as isn't `AUTO_INCREMENT`, `LOCK=NONE` is
-  allowed
-- Renaming a column: don’t rename, instead
-  - `vN`: non breaking, add a new column with `LOCK=NONE`, allow `NULL` values.
-  - `vN+1`: stop writing to the column.
-  - `vN+2`: breaking, remove the old column, add `NOT NULL` constraints if
-    necessary.
+- Adding a new column: append `LOCK=NONE` to the `ALTER` statement (except for
```

Comment:
I would mention something about avoiding use of `select *` in queries. Adding a new column breaks old fleet versions if they use `*` during scanning ie "missing destination"

Reply:
very good catch, thanks!